plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -74,5 +99,206 @@ public function delete_database_tables() {
 		}
 	}
 
+	/**
+	 * Is the data sync between old and new tables currently enabled?
+	 *
+	 * @return bool
+	 */
+	public function data_sync_is_enabled(): bool {
+		return 'yes' === get_option( self::ORDERS_DATA_SYNC_ENABLED_OPTION );
+	}
+
+	/**
+	 * Is a sync process currently in progress?
+	 *
+	 * @return bool
+	 */
+	public function pending_data_sync_is_in_progress(): bool {
+		return 'yes' === get_option( self::PENDING_SYNC_IS_IN_PROGRESS_OPTION );
+	}
+
+	/**
+	 * Get the current sync process status.
+	 * The information is meaningful only if pending_data_sync_is_in_progress return true.
+	 *
+	 * @return array
+	 */
+	public function get_sync_status() {
+		return array(
+			'initial_pending_count' => (int) get_option( self::INITIAL_ORDERS_PENDING_SYNC_COUNT_OPTION, 0 ),
+			'current_pending_count' => $this->get_current_orders_pending_sync_count(),
+			'auto_flip'             => 'yes' === get_option( self::AUTO_FLIP_AUTHORITATIVE_TABLE_ROLES_OPTION ),
+			'sync_in_progress'      => $this->pending_data_sync_is_in_progress(),
+		);
+	}
+
+	/**
+	 * Calculate how many orders need to be synchronized currently.
+	 *
+	 * If an option whose name is given by self::FAKE_ORDERS_PENDING_SYNC_COUNT_OPTION exists,
+	 * then the value of that option is returned. This is temporary, to ease testing the feature
+	 * while it is in development.
+	 *
+	 * Otherwise a database query is performed to get how many orders match one of the following:
+	 *
+	 * - Existing in the authoritative table but not in the backup table.
+	 * - Existing in both tables, but they have a different update date.
+	 */
+	public function get_current_orders_pending_sync_count(): int {
+		global $wpdb;
+
+		// TODO: Remove the usage of the fake pending orders count once development of the feature is complete.
+		$count = get_option( self::FAKE_ORDERS_PENDING_SYNC_COUNT_OPTION );
+		if ( false !== $count ) {
+			return (int) $count;
+		}
+
+		$orders_table = $wpdb->prefix . 'wc_orders';
+
+		if ( 'yes' === get_option( CustomOrdersTableController::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION ) ) {
+			$missing_orders_count_sql = "SELECT COUNT(1) FROM $orders_table WHERE post_id IS NULL";
+		} else {
+			$missing_orders_count_sql = "
+SELECT COUNT(1) FROM (
+	SELECT posts.ID,orders.post_id FROM $wpdb->posts posts
+	LEFT JOIN $orders_table orders ON posts.ID = orders.post_id
+	WHERE posts.post_type='shop_order' AND orders.post_id IS NULL
+	AND posts.post_status != 'auto-draft'
+) x";
```

Comment:
Can this be simplified?

```suggestion
SELECT COUNT( posts.ID ) FROM $wpdb->posts posts
LEFT JOIN $orders_table orders ON posts.ID = orders.post_id
WHERE posts.post_type='shop_order' AND orders.post_id IS NULL
AND posts.post_status != 'auto-draft'
```